restli-tools/src/main/java/com/linkedin/restli/tools/clientgen/fluentspec/RestMethodSpec.java

```diff
@@ -124,4 +131,43 @@ public int getQueryParamMapSize()
     }
     return DataMapBuilder.getOptimumHashMapCapacityFromSize(params);
   }
+
+  public Set<ProjectionParameterSpec> getSupportedProjectionParams()
+  {
+    switch (ResourceMethod.fromString(_schema.getMethod()))
+    {
+      case GET:
+      case BATCH_GET:
+        return Collections.singleton(new ProjectionParameterSpec(RestConstants.FIELDS_PARAM, FIELDS_MASK_METHOD_NAME, _root.getEntityClass()));
+      case CREATE:
+      case BATCH_CREATE:
+      case PARTIAL_UPDATE:
+      case BATCH_PARTIAL_UPDATE:
+        if (returnsEntity())
+        {
+          return Collections.singleton(new ProjectionParameterSpec(RestConstants.FIELDS_PARAM, FIELDS_MASK_METHOD_NAME, _root.getEntityClass()));
+        }
+        else
+        {
+          return Collections.emptySet();
+        }
+      case FINDER:
+      case BATCH_FINDER:
+      case GET_ALL:
+        Set<ProjectionParameterSpec> collectionParts = new HashSet<>();
+        collectionParts.add(new ProjectionParameterSpec(RestConstants.FIELDS_PARAM, FIELDS_MASK_METHOD_NAME, _root.getEntityClass()));
+        if (_schema.getMetadata() != null)
+        {
+          collectionParts.add(new ProjectionParameterSpec(RestConstants.METADATA_FIELDS_PARAM, METADATA_MASK_METHOD_NAME, _root.classToTemplateSpec(_schema.getMetadata().getType())));
+        }
+        if (_schema.hasPagingSupported() && _schema.isPagingSupported())
+        {
+          // TODO: add suppport for paging projection.
```

Comment:
Is this in the roadmap?

Reply:
This will be done along with finder support and we decide how to return paging data.